packages/amplify-graphql-transformer-migrator/src/schema-migrator.ts

```diff
@@ -0,0 +1,193 @@
+    return [];
+  }
+  if (schemaFileExists) {
+    return await readSingleFileSchema(schemaFilePath);
+  }
+  else if (schemaDirectoryExists) {
+    return await readSchemaDocuments(schemaDirectoryPath);
+  }
+  return [];
+}
+
+async function runMigration(schemas: SchemaDocument[], authMode: string): Promise<boolean> {
+  let fullSchema: string;
+  let schemaList: Array<string> = new Array<string>();
+  let backupSchemaPromiseArray: Array<Promise<void>> = new Array<Promise<void>>();
+  schemas.forEach((doc) => {
+    schemaList.push(doc.schema);
+    backupSchemaPromiseArray.push(backupSchema(doc.filePath));
+  });
+  await Promise.all(backupSchemaPromiseArray);
+
+  fullSchema = schemaList.join('\n');
+  let fullSchemaNode = parse(fullSchema);
+  doSchemaValidation(fullSchema);
+
+  let newSchemaList: SchemaDocument[] = new Array<SchemaDocument>();
+  for(let doc of schemas) {
+    const newSchema = await migrateGraphQLSchema(doc.schema, authMode, fullSchemaNode);
+    newSchemaList.push({ schema: newSchema, filePath: doc.filePath });
+  }
+
+  const diffs = getSchemaDiffs(schemas, newSchemaList);
+  showDiffs(diffs);
+
+  const migrationChoices: Array<string> = ["Yes", "Yes, but exit the CLI so I can review/edit my new schemas before compiling/pushing",
+    "No, continue my operation with my old schemas"];
+  const migrationChoice: string = await prompter.pick("Do you want to proceed with the auto-migration?", migrationChoices);
+  const migrationIndex: number = migrationChoices.findIndex((value) => {
+    return value === migrationChoice;
+  });
+  if (migrationIndex === -1 || migrationIndex === 2) {
+    return false;
+  }
+
+  let writeNewPromiseArray: Promise<void>[] = [];
+  newSchemaList.forEach((doc) => {
+    writeNewPromiseArray.push(replaceFile(doc.schema, doc.filePath));
+  });
+  await Promise.all(writeNewPromiseArray);
+
+  if (migrationIndex === 1) {
+    // Customer has selected to review their new schemas before compiling or pushing
+    exitOnNextTick(0);
+  }
+  printer.info(`Success! Auto-migration completed. Just in case, we've placed a back-up of your old 
+GraphQL schema file(s) at these location(s):`);
+  for(let schema of schemas) {
+    printer.info(`- ${schema.filePath}.bkp`);
+  }
+  return true;
+}
+
+async function migrateToV2Transformer(resourceDir: string, context: $TSContext, schemaDocs: SchemaDocument[]): Promise<boolean> {
+  const { envName } = context.amplify.getEnvInfo();
+  const defaultAuth = await getDefaultAuthFromContext();
+  const authMode = cliToMigratorAuthMap.get(defaultAuth);
+  if (!authMode) {
+    throw Error(`Unidentified authorization mode for API found: ${defaultAuth}`);
+  }
+
+  try {
+    await runMigration(schemaDocs, authMode);
+    await updateTransformerVersion(envName);
+  }
+  catch (error) {
+    printer.error("Encountered an error while migrating schema to V2 transformer, reverting to old schemas");
+    await undoAllSchemaMigration(resourceDir);
```

Comment:
Is it possible for this to throw? If so, it should be wrapped in its own `try...catch`. If it throws, we may want to surface that error as well.